x-pack/plugins/security/server/authentication/authenticator.ts

```diff
@@ -754,4 +756,30 @@ export class Authenticator {
         })
       : authenticationResult;
   }
+
+  /**
+   * Creates a logged out URL for the specified request and provider.
+   * @param request Request that initiated logout.
+   * @param providerType Type of the provider that handles logout.
+   */
+  private getLoggedOutURL(request: KibanaRequest, providerType: string) {
+    // The app that handles logout needs to know the reason of the logout and the URL we may need to
+    // redirect user to once they log in again (e.g. when session expires).
+    const searchParams = new URLSearchParams();
+    for (const [key, defaultValue] of [
+      [NEXT_URL_QUERY_STRING_PARAMETER, null],
+      [LOGOUT_REASON_QUERY_STRING_PARAMETER, 'LOGGED_OUT'],
+    ] as Array<[string, string | null]>) {
+      const value = request.url.searchParams.get(key) || defaultValue;
+      if (value) {
+        searchParams.append(key, value);
+      }
+    }
+
+    // Query string may contain the path where logout has been called or
+    // logout reason that login page may need to know.
+    return this.options.config.authc.selector.enabled || shouldProviderUseLoginForm(providerType)
+      ? `${this.options.basePath.serverBasePath}/login?${searchParams.toString()}`
+      : `${this.options.basePath.serverBasePath}/security/logged_out?${searchParams.toString()}`;
```

Comment:
__note:__ `logged_out` should support not only `next`, but also `msg` to eventually display different error messages for the normal logout and the logout caused by the session timeout https://github.com/elastic/kibana/issues/84200.